src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployService.kt

```diff
@@ -31,102 +28,64 @@ class RedeployService(val openShiftClient: OpenShiftClient, val openShiftObjectG
         }
     }
 
-    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, openShiftResponses: List<OpenShiftResponse>): RedeployResult {
+    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, redeployContext: RedeployContext): RedeployResult {
+        val redeployResourceFromSpec = generateRedeployResourceFromSpec(deploymentSpec, redeployContext) ?: return RedeployResult()
```

Comment:
Here we can get back a DeployRequest. If we do there is really no need to run all the code below. What if we create a type out of the response here and do some switching on that? Then it is also easier to give a good message to the user. 

IE if we have a development flow we want to say so in the message=No deploy was made with development type.

Reply:
I have moved the check for type build or development into the triggerRedeploy method and added a better response message if this case happens. Also added unit test for it.